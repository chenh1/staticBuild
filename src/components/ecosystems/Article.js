import React from 'react';
import { Link, IndexLink } from 'react-router';
import { TextBlock } from '../organisms';
import { ImageContainer } from '../molecules';

const Article = ({articleItems}) => {
    return (
        <div className="standard-article">
            {articleItems.map((item, index) => {
                switch (item.type) {
                    case 'image':
                        return (
                            <ImageContainer 
                                key={"item" + index}
                                customClass={item.customClass}
                                imageSrc={item.imageSrc}
                                alt={item.alt}
                                caption={item.caption}
                            />
                        );
                    case 'text':
                        return (
                            <TextBlock 
                                key={"item" + index}
                                elements={item.content}
                            />
                        );
                    default:
                        return;
                }
            })}
        </div>
    );
};

export default Article;