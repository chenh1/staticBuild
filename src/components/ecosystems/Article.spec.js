import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Article from './Article';

test('renders overarching article block with images and texts', () => {
    const tree = renderer.create(<Article
        articleItems={[
            {type: 'image', customClass: 'mock-class', imageSrc: '/mockSrc', alt: 'mock alt', caption: 'Mock caption'},
            {type: 'text', content: [
                {type: 'paragraph', text: ['Mock paragraph text', '...more text']},
                {type: 'subHeader', text: 'Mock subheader text'},
                {type: 'bulletList', listItems: ['list item 1', 'list item 2']}
            ]}
        ]} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});