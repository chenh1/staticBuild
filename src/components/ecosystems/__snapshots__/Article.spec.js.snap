// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders overarching article block with images and texts 1`] = `
<div
  className="standard-article"
  id="articleContent"
>
  <div
    className="mock-class"
  >
    <img
      alt="mock alt"
      src="/mockSrc"
    />
    <span
      className="image-caption"
    >
      Mock caption
    </span>
  </div>
  <div
    className="text-wrapper"
  >
    <p
      className="paragraph-one"
    >
      Mock paragraph text
      ...more text
    </p>
    <h3
      className="header-three"
    >
      Mock subheader text
    </h3>
    <ul
      className="list-one"
    >
      <li>
        <span>
          list item 1
        </span>
      </li>
      <li>
        <span>
          list item 2
        </span>
      </li>
    </ul>
  </div>
</div>
`;
